src/base/random.h

```diff
@@ -47,8 +47,8 @@ struct RNG {
 	}
 
 private:
-	uint32_t state0;
-	uint32_t state1;
+	uint32_t state0 = 0;
```

Comment:
why don't we use the braces notation here? Furthermore these are unsigned

Reply:
C++ is mixing two different kinds of member initializers now. The `=` operator is a relatively old (pre-C++17) feature that allows us to assign a literal value to a member variable in the default constructor. This only works if the literal value is of the same data type as the variable, e.g. integer or (null-)pointer.

The `{braces}` syntax is an initializer list, which is a new feature in C++17. It is equivalent to calling the object constructor in the constructor's initializer list (which in turn is new in C++11). So this is actually an argument list to pass to a class constructor.

For class types, the second type is more powerful. For example:
```c++
Time t{5};
```
calls the `Time(uint32_t)` constructor with argument `5`. On the other hand
```c++
Time t = 5;
```
is a compile error because `5` is of type `int` and not of type `Time`.

For primitive types such as integers though, both ways are equivalent and it doesn't really matter, so I changed it to braces notation for consistency.

Reply:
thanks. my remark was indeed targeted at consistency